crates/consensus/src/validation.rs

```diff
@@ -353,12 +354,25 @@ pub fn validate_block_regarding_chain<PROV: HeaderProvider>(
         .header(&block.parent_hash)?
         .ok_or(Error::ParentUnknown { hash: block.parent_hash })?;
 
+    // Check if withdrawals are valid.
+    let parent_block_id = BlockId::Number(BlockNumberOrTag::Number(parent.number));
+    let latest_withdrawal = provider.lastest_withdrawal(parent_block_id, parent.timestamp)?;
+    if let Some(withdrawals) = &block.withdrawals {
+        if !withdrawals.is_empty() {
+            let withdrawal_index = withdrawals.first().unwrap().index;
+            let lastest_withdrawal_index = latest_withdrawal.map_or(0, |w| w.index);
+            if lastest_withdrawal_index + 1 != withdrawal_index {
```

Comment:
the first withdrawal should have index 0. this logic would fail, because it expects index 1 if no latest withdrawal was found